src/gbt_model_builder.pyx

```diff
@@ -179,3 +181,110 @@ def gbt_reg_model_builder(n_features, n_iterations):
+        self.daal_model_ = get_gbt_model_from_catboost(booster)
+        self._get_params_from_catboost(catboost_params)
+
+
+    def build_model(self, model):
+        (submodule_name, class_name) = (model.__class__.__module__,
+                                        model.__class__.__name__)
+        # Build from LightGBM
+        if (submodule_name, class_name) == ("lightgbm.basic", "Booster"):
+            self._build_model_from_lightgbm(model)
+        # Build from XGBoost
+        elif (submodule_name, class_name) == ("xgboost.core", "Booster"):
+            self._build_model_from_xgboost(model)
+        # Build from CatBoost
+        elif (submodule_name, class_name) == ("catboost.core", "CatBoost"):
+            self._build_model_from_catboost(model)
+        else:
+            raise TypeError(f"Unknown model format {submodule_name}.{class_name}")
+
+        self.is_regression_ = isinstance(self.daal_model_, gbt_regression_model)
+
+        return self
+
+    def _predict_classification(self, X, fptype, resultsToEvaluate):
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError('Shape of input is different from what was seen in `fit`')
+
+        if not hasattr(self, 'daal_model_'):
+            raise ValueError((
+                "The class {} instance does not have 'daal_model_' attribute set. "
+                "Call 'fit' with appropriate arguments before using this method.").format(
+                    type(self).__name__))
+
+        # Prediction
+        predict_algo = gbt_classification_prediction(
+            fptype=fptype,
+            nClasses=self.n_classes_,
+            resultsToEvaluate=resultsToEvaluate)
+        predict_result = predict_algo.compute(X, self.daal_model_)
+
+        if resultsToEvaluate == "computeClassLabels":
+            return predict_result.prediction.ravel().astype(np.int64, copy=False)
+        else:
+            return predict_result.probabilities
+
+    def _predict_regression(self, X, fptype):
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError('Shape of input is different from what was seen in `fit`')
+
+        if not hasattr(self, 'daal_model_'):
+            raise ValueError((
+                "The class {} instance does not have 'daal_model_' attribute set. "
+                "Call 'fit' with appropriate arguments before using this method.").format(
+                    type(self).__name__))
+
+        # Prediction
+        predict_algo = gbt_regression_prediction(fptype=fptype)
+        predict_result = predict_algo.compute(X, self.daal_model_)
+
+        return predict_result.prediction.ravel()
+
+    def predict(self, X, fptype="float", resultsToEvaluate="computeClassLabels"):
+        if self.is_regression_:
+            return self._predict_regression(X, fptype)
+        else:
+            return self._predict_classification(X, fptype, resultsToEvaluate)
+
+def gbt_build_model(model):
+    return GBTDAALModel().build_model(model)
```

Comment:
Why do not use static method in GBTDAALModel instead of separate function?

Reply:
> Why do not use static method in GBTDAALModel instead of separate function?

Separate function is less wordily and looks more clear for the end user.

Reply:
It's not obvious what is returned from this function while static method will make association with estimator class
Also, `gbt_build_model` and `GBTDAALModel` use abbreviature while sklearn API/docs prefer [full name](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html#sklearn-ensemble-gradientboostingclassifier).